{{cookiecutter.project_slug}}/.pre-commit-config.yaml

```
@@ -1,57 +1,82 @@
+  - id: check-added-large-files
+  - id: check-executables-have-shebangs
+  - id: check-case-conflict
+  - id: check-vcs-permalinks
+  - id: forbid-new-submodules
+  - id: mixed-line-ending
+  - id: check-merge-conflict
+  - id: detect-private-key
+  - id: detect-aws-credentials
+    args: ['--allow-missing-credentials']  # Avoid failure on CI
+  - id: check-toml
+  - id: check-yaml
+  - id: check-json
+- repo: https://github.com/shellcheck-py/shellcheck-py
+  rev: v0.8.0.4
+  hooks:
+  - id: shellcheck
+    name: Shellcheck
+    args: ["-f", "gcc"]  # output filename:linenum:colnum (clickable)
+- repo: https://github.com/AleksaC/hadolint-py
+  rev: v2.10.0
+  hooks:
+  - id: hadolint
+    name: Hadolint (Dockerfile checker)
 # Actual Python Linters
--   repo: https://github.com/psf/black
-    rev: 22.10.0
-    hooks:
-      - id: black
-        name: Black (Python formatter)
--   repo: https://github.com/pycqa/isort
-    rev: 5.10.1
-    hooks:
-      - id: isort
-        name: isort (Python import sorter)
--   repo: https://github.com/pycqa/flake8
-    rev: 5.0.4
-    hooks:
-    -   id: flake8
-        name: Flake8 (Python linter)
-        args: ["--output-file", "test_results/flake8.txt", "--tee", "src/", "tests/"]
-        additional_dependencies:
-          - "flake8-bandit==4.1.1"
-          - "flake8-bugbear==22.8.23"
-          - "flake8-docstrings==1.6.0"
--   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.982
-    hooks:
-    -   id: mypy
-        name: Mypy (Python type-checker)
+- repo: https://github.com/psf/black
+  rev: 22.12.0
+  hooks:
+  - id: black
+    name: Black (Python formatter)
+- repo: https://github.com/pycqa/isort
+  rev: v5.11.3
+  hooks:
+  - id: isort
+    name: isort (Python import sorter)
+- repo: https://github.com/pycqa/flake8
+  rev: 6.0.0
+  hooks:
+  - id: flake8
+    name: Flake8 (Python linter)
+    # flake8 refuses to use pyproject.toml, needs a separate config.
+    # See https://github.com/PyCQA/flake8/issues/234
+    args: [
+      "--config=.flake8",
+      "--output-file",
+      "test_results/flake8.txt",
+      "--tee",
+      "src/",
+      "tests/"
+    ]
+    additional_dependencies:
+    # Bandit is currently disabled it because doesn't seem to respect
+    # per-file-ignores=tests:S101 and keeps complaining about asserts
+    # in tests.
+    # - "flake8-bandit==4.1.1"
+    - "flake8-bugbear==22.8.23"
+    - "flake8-docstrings==1.6.0"
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: v0.991
+  hooks:
+  - id: mypy
+    name: Mypy (Python type-checker)
+    # Uncomment below if mypy requires extra type stub packages
+    # additional_dependencies: [types-PyYAML==6.0.12.2]
+- repo: https://github.com/igorshubovych/markdownlint-cli
+  rev: v0.32.2
+  hooks:
+  - id: markdownlint
+    args: ["--config", ".markdown-lint.yml"]
+- repo: https://github.com/adrienverge/yamllint.git
+  rev: v1.17.0
+  hooks:
+  - id: yamllint
+    args: ["-c", ".yamllint"]
```

To avoid file clutter, we can kill the yamllint config file, especially since config file is currently just a single key.

Suggest we follow [setting config options without config file](https://yamllint.readthedocs.io/en/stable/configuration.html#custom-configuration-without-a-config-file):

```suggestion
    args: ["-d", "{rules: {indentation: {indent-sequences: false}}}"]
```
Untested the particular dict, but docs says it should work.

---

Removed the config file and added inline config instead. I actually noticed that my config was slightly incorrect: it would define just a single rule that was used, instead of extending the default rule set. Also relaxed the rule slightly